src/libraries/Microsoft.Extensions.DependencyInjection/src/Microsoft.Extensions.DependencyInjection.csproj

```diff
@@ -24,9 +24,8 @@
              Link="Common\src\Extensions\ParameterDefaultValue\ParameterDefaultValue.cs" />
     <Compile Include="$(CommonPath)Extensions\TypeNameHelper\TypeNameHelper.cs"
              Link="Common\src\Extensions\TypeNameHelper\TypeNameHelper.cs" />
-  </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' != '$(NetCoreAppCurrent)'">
+    <!-- These types weren't available before net5.0 and need to be compiled into the assembly. -->
```

Comment:
This will break trimming removal for the types and they are useless for pre-net5 configs

Reply:
We will need to fix the ILLink.LinkAttributes.xml file to tell the linker to remove these attributes from any assembly.

https://github.com/dotnet/runtime/blob/ada53f0c05efd882e29e8f225a483521f08e3caa/src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml#L18

Just like we do for:

https://github.com/dotnet/runtime/blob/ada53f0c05efd882e29e8f225a483521f08e3caa/src/mono/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.xml#L145-L156

These attributes are allowed to be in any assembly. It doesn't make sense to make every library to build for net5.0+ just so it can reference the attributes from corelib.

cc @vitek-karas

Reply:
Sure we can fix the attribute scope but I'm not sure this change is a step in the right direction. Sooner or later there will be net5+ API which one of the assemblies could benefit from and we will need to undo this change.

Reply:
> Sooner or later there will be net5+ API which one of the assemblies could benefit from and we will need to undo this change.

It's fine to undo the #ifdef change in the future when we want to depend on a newer API set. The `$(NetCoreAppCurrent)` configuration which I removed wasn't shipping and was only building because of a previous limitation in the build system.